src/components/UserProfile/UserProfileEdit/UserProfileEdit.jsx

```diff
@@ -0,0 +1,641 @@
+import React, { Component } from 'react'
+import Loading from '../../common/Loading'
+import {
+	Row,
+	Label,
+	Input,
+	CardTitle,
+	Col,
+	Container,
+	Button,
+	Badge,
+	Collapse,
+} from 'reactstrap'
+
+import Image from 'react-bootstrap/Image'
+import { orange, warningRed } from '../../../constants/colors'
+import BlueSquare from '../BlueSquares'
+import Modal from '../UserProfileModal'
+import UserLinks from '../UserLinks'
+import ToggleSwitch from './ToggleSwitch'
+
+import styleProfile from '../UserProfile.css'
+import styleEdit from './UserProfileEdit.css'
+
+
+class EditProfile extends Component {
+	state = {
+		isLoading: true,
+		error: '',
+		userProfile: {},
+		firstNameError: '',
+		lastNameError: '',
+		imageUploadError: '',
+		isValid: false,
+		id: '',
+		privacySettings:{
+			email: true,
+			phoneNumber: true,
+			blueSquares: true
+		}
+	}
+
+	async componentDidMount() {
+		// this.props.getCurrentUser(getjwt())
+		if (this.props.match) {
+			let userId = this.props.match.params.userId
+			await this.props.getUserProfile(userId)
+			await this.props.getUserTeamMembers(userId)
+			//console.log(this.props.userProfile)
+			if (this.props.userProfile.firstName.length) {
+				// check props:
+				console.log('props:', this.props.userProfile)
+				// fill in defaults where needed
+				this.setState({ isLoading: false, userProfile: this.props.userProfile })
+
+				if (this.props.userProfile.privacySettings){
+					this.setState({
+						isLoading: false,
+						userProfile:{
+							...this.props.userProfile
+						}
+					})
+				}
+
+			}
+		}
+	}
+
+	handleUserProfile = event => {
+
+		if (event.target.id === 'firstName') {
+			this.setState({
+				userProfile: {
+					...this.state.userProfile,
+					firstName: event.target.value.trim()
+				},
+				showModal: event.target.value ? false : true,
+				modalTitle: 'First Name Error',
+				modalMessage: 'First Name cannot be empty'
+			})
+		}
+
+		if (event.target.id === 'lastName') {
+			this.setState({
+				userProfile: {
+					...this.state.userProfile,
+					lastName: event.target.value.trim()
+				},
+				showModal: event.target.value ? false : true,
+				modalTitle: 'First Name Error',
+				modalMessage: 'Last Name cannot be empty'
+			})
+		}
+
+		if (event.target.id === 'jobTitle') {
+			this.setState({
+				userProfile: {
+					...this.state.userProfile,
+					jobTitle: event.target.value
```

Comment:
When I see the jobTitle from my mockedState file it looks to be an array.  This may be the issue with jobTitle not updating not sure.